references/cli/workflow.go

```diff
@@ -246,6 +249,91 @@ func NewWorkflowDebugCommand(c common.Args, ioStream cmdutil.IOStreams, wargs *W
 	return cmd
 }
 
+// NewWorkflowListCommand create workflow list command
+func NewWorkflowListCommand(c common.Args, ioStream cmdutil.IOStreams, wargs *WorkflowArgs) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List running workflows",
+		Long:    "List running workflows",
+		Example: "vela workflow list",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cli, err := c.GetClient()
+			if err != nil {
+				return err
+			}
+			namespace, err := GetFlagNamespaceOrEnv(cmd, c)
+			if err != nil {
+				return err
+			}
+			if AllNamespace {
+				namespace = ""
+			}
+			ctx := context.Background()
+			return printWorkflowList(ctx, cli, namespace, ioStream)
+		},
+	}
+	cmd.Flags().BoolVarP(&AllNamespace, "all-namespaces", "A", false, "If true, check the specified action in all namespaces.")
+	addNamespaceAndEnvArg(cmd)
+	return cmd
+}
+
+func printWorkflowList(ctx context.Context, c client.Reader, namespace string, ioStream cmdutil.IOStreams) error {
+	table, err := buildWorkflowListTable(ctx, c, namespace)
+	if err != nil {
+		return err
+	}
+	ioStream.Info(table.String())
+	return nil
+}
+
+func buildWorkflowListTable(ctx context.Context, c client.Reader, namespace string) (*uitable.Table, error) {
+	table := newUITable()
+	header := []interface{}{"NAME", "TYPE", "PHASE", "START-TIME", "END-TIME"}
+	if AllNamespace {
+		header = append([]interface{}{"NAMESPACE"}, header...)
+	}
+	table.AddRow(header...)
+	applist := v1beta1.ApplicationList{}
+	if err := c.List(ctx, &applist, client.InNamespace(namespace)); err != nil {
+		if apierrors.IsNotFound(err) {
+			return table, nil
+		}
```

Comment:
Will `List` return not found error? If it truly does, we should also continue to fetch workflowRuns

Reply:
Thats fair @chivalryq . Let me take a look . cc: @muralicharan